metadata-ingestion/src/datahub/ingestion/source/dataplex/dataplex_config.py

```diff
@@ -0,0 +1,202 @@
+        "Extracts table-level lineage relationships between entities.",
+    )
+
+    lineage_fail_fast: bool = Field(
+        default=False,
+        description="If True, fail immediately on lineage extraction errors. "
+        "If False (default), log errors and continue processing other entities. "
+        "Recommended: False for development, True for production to catch issues early.",
+    )
+
+    max_lineage_failures: int = Field(
+        default=10,
+        description="Maximum number of consecutive lineage extraction failures before stopping. "
+        "Acts as a circuit breaker to prevent excessive API calls when there's a systemic issue. "
+        "Set to -1 to disable circuit breaker. Default: 10.",
+    )
+
+    lineage_batch_size: int = Field(
+        default=1000,
+        description="Number of entities to process in each lineage extraction batch. "
+        "Lower values reduce memory usage but may increase processing time. "
+        "Set to -1 to disable batching (process all entities at once). "
+        "Recommended: 1000 for large deployments, -1 for small deployments. Default: 1000.",
+    )
+
+    create_sibling_relationships: bool = Field(
+        default=True,
+        description="Whether to create sibling relationships between Dataplex entities and native platform datasets (BigQuery/GCS).",
+    )
+
+    dataplex_is_primary_sibling: bool = Field(
+        default=False,
+        description="Controls sibling relationship primary designation between Dataplex entities and source platform entities. "
+        "When False (default), source platform entities (BigQuery/GCS) are primary and Dataplex entities are secondary. "
+        "When True, Dataplex entities are primary and source platform entities are secondary. "
+        "Default is False since source platforms are typically the canonical data source.",
+    )
+
+    apply_zone_type_tags: bool = Field(
+        default=True,
+        description="Whether to apply tags based on zone types (RAW, CURATED).",
+    )
+
+    apply_label_tags: bool = Field(
+        default=True,
+        description="Whether to convert Dataplex labels to DataHub tags.",
+    )
+
+    max_workers: int = Field(
+        default=10,
+        description="Number of worker threads to use to parallelize zone entity extraction. Set to 1 to disable parallelization.",
+    )
+
+    dataplex_url: str = Field(
+        default="https://console.cloud.google.com/dataplex",
+        description="Base URL for Dataplex console (for generating external links).",
+    )
+
+    @model_validator(mode="before")
+    @classmethod
+    def project_id_backward_compatibility(cls, values: Dict) -> Dict:
+        """Handle backward compatibility for project_id -> project_ids migration."""
+        # Create a copy to avoid modifying the input dictionary, preventing state contamination in tests
+        values = deepcopy(values)
```

Comment:
This was added during some pydantic v1->v2 refactor because some tests were reusing dicts across tests. I think that's a bad practice. But at that moment of the refactor I didn't want to touch the tests, so I did the `deepcopy`.

My suggestion would be to prevent tests to reuse config dicts so the `deepcopy` is not needed.

Minor comment.

Reply:
Agree, made changes:

1. Removed from copy import deepcopy
2. Changed the validator to create a new dict using dict comprehension: {k: v for k, v in values.items() if k != "project_id"}
3. Verified with tests that:
 - project_id -> project_ids migration works
 - Dict reuse doesn't cause issues (original dict is preserved)
 - Both project_id and project_ids set works correctly
 - Normal project_ids usage works